### PR TITLE
Enhance Inference with Gradient Freezing and Eval Mode

### PR DESCRIPTION
**Pull request by Brikarl**
Introduced two updates to improve inference efficiency: 

1. Gradient Freezing (`base_model.requires_grad_(False)`) to prevent unnecessary computations by freezing gradients, reducing memory use and speeding up the model. 
2. Evaluation Mode (`base_model.eval()`) ensures the model operates in inference mode, disabling training-specific layers like Dropout for consistent results. 

These changes streamline performance and are essential for deploying the model in production environments.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -164,4 +164,5 @@ cython_debug/
 nohup.out
 transformers/models/
 peft/models/
-peft/temp/
\ No newline at end of file
+peft/temp/
+hf_transformers/
```
Comment (DjangoPeng) [inline .gitignore]: 请更新下代码，这个目录已经重新命名了。

File: peft/peft_chatglm_inference.ipynb [hunk 2]
```diff
@@ -56,7 +56,9 @@
     "base_model = AutoModel.from_pretrained(config.base_model_name_or_path,\n",
     "                                       quantization_config=q_config,\n",
     "                                       trust_remote_code=True,\n",
-    "                                       device_map='auto')"
+    "                                       device_map='auto')\n",
+    "base_model.requires_grad_(False)\n",
```
Comment (DjangoPeng) [inline peft/peft_chatglm_inference.ipynb:60]: 单独提交这个作为 PR 是没问题的。

**Comment by DjangoPeng** [comment]
LGTM
